src/arch/x86_64/CodeGen.zig

```diff
@@ -6314,18 +6347,17 @@ fn resolveInst(self: *Self, inst: Air.Inst.Ref) InnerError!MCValue {
             return gop.value_ptr.*;
         },
         .const_ty => unreachable,
-        else => return self.getResolvedInstValue(inst_index),
+        else => return self.getResolvedInstValue(inst_index).?,
     }
 }
 
-fn getResolvedInstValue(self: *Self, inst: Air.Inst.Index) MCValue {
+fn getResolvedInstValue(self: *Self, inst: Air.Inst.Index) ?MCValue {
     // Treat each stack item as a "layer" on top of the previous one.
     var i: usize = self.branch_stack.items.len;
     while (true) {
         i -= 1;
         if (self.branch_stack.items[i].inst_table.get(inst)) |mcv| {
-            assert(mcv != .dead);
-            return mcv;
+            return if (mcv != .dead) mcv else null;
```

Comment:
Could you explain why this is needed? I thought the idea was to never permit `.dead` MCVs to linger in nested blocks.

Reply:
I have converted the assertion condition to a `null` return and then added `.?` to most call sites to preserve the assertion.  If there were more users, or it was likely to be called in new air functions, I would have split into two functions instead.  As for the single use that ignores the assertion, this is because prong deaths are not guaranteed to be unique, so if a dead mcv

Reply:
Are you referring to double `%0! %0!` in the following AIR for example?

```
# Begin Function AIR: twice.f:                                                                                                                                 
# Total AIR+Liveness bytes: 675B                                                                                                                               
# AIR Instructions:         35 (315B)                                                                                                                          
# AIR Extra Data:           33 (132B)                                                                                                                          
# AIR Values Bytes:         8 (64B)                                                                                                                            
# Liveness tomb_bits:       24B                                                                                                                                
# Liveness Extra Data:      5 (20B)                                                                                                                            
# Liveness special table:   1 (8B)                                                                                                                             
  %6 = constant(u1, 0)                                                                                                                                         
  %7 = constant(u1, 1)                                                                                                                                         
  %18 = const_ty(*const type)                                                                                                                                  
  %19 = constant(*const type, builtin)                                                                                                                         
  %20 = constant(type, builtin)                                                                                                                                
  %21 = const_ty(*const type)                                                                                                                                  
  %22 = constant(*const type, builtin.panic_messages)                                                                                                          
  %23 = constant(type, builtin.panic_messages)                                                                                                                 
  %24 = const_ty(*const *const [23:0]u8)                                                                                                                       
  %25 = constant(*const *const [23:0]u8, "switch on corrupt value")                                                                                            
  %26 = const_ty(*const [23:0]u8)                                                                                                                              
  %27 = constant(*const [23:0]u8, "switch on corrupt value")                                                                                                   
                                                                                                                                                               
  %0 = arg(u1, 0)                                                                                                                                              
  %2!= dbg_block_begin()                                                                                                                                       
  %3!= dbg_stmt(2:5)                                                                                                                                           
  %4 = ret_ptr(*u1)                                                                                                                                            
  %5!= dbg_stmt(2:20)                                                                                                                                          
  %8!= block(void, {                                                                                                                                           
    %30!= switch_br(%0, [%6] => {                                                                                                                              
        %9!= dbg_block_begin()                                                                                                                                 
        %10!= store(%4, %0!)                                                                                                                                   
        %11!= dbg_block_end()                                                                                                                                  
        %12!= br(%8, @Zir.Inst.Ref.void_value)
      }, [%7] => {
        %13!= dbg_block_begin()
        %14!= store(%4, %0!)
        %15!= dbg_block_end()
        %16!= br(%8, @Zir.Inst.Ref.void_value)
      }, else => {
        %0! %0!
        %17!= dbg_stmt(2:20)
        %28!= breakpoint()
        %29!= unreach()
      }
    })
  })
  %31!= dbg_stmt(2:5)
  %32!= load(u1, %4)
  %34!= dbg_block_end()
  %33!= ret_load(%4!)
# End Function AIR: twice.f
```

Reply:
Yep, that's it.  You can see the assertion trigger my compiling that code without these changes.